azurerm/resource_arm_container_group.go

```diff
@@ -531,6 +567,63 @@ func expandContainerEnvironmentVariables(input interface{}) *[]containerinstance
 	return &output
 }
 
+func expandContainerImageRegistryCredentials(d *schema.ResourceData) *[]containerinstance.ImageRegistryCredential {
+	credsRaw := d.Get("image_registry_credential").([]interface{})
+	if len(credsRaw) == 0 {
+		return nil
+	}
+
+	output := make([]containerinstance.ImageRegistryCredential, 0, len(credsRaw))
+
+	for _, c := range credsRaw {
+		credConfig := c.(map[string]interface{})
+
+		server := credConfig["server"].(string)
+		username := credConfig["username"].(string)
+		password := credConfig["password"].(string)
+
+		output = append(output, containerinstance.ImageRegistryCredential{
+			Server:   &server,
```

Comment:
Very minor but these could be assigned inline via `utils.String()`

```go
Server: utils.String(credConfig["username"].(string)),
```

Reply:
Ah much nicer, I thought about bringing in the `to` package from Autorest as has lots of these funcs too didn't realize you already had one.